lib/pbench/test/unit/server/query_apis/test_datasets_contents.py

```diff
@@ -0,0 +1,412 @@
+                                }
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                    {
+                        "_index": "riya-pbench.v6.run-toc.2021-05",
+                        "_type": "_doc",
+                        "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                        "_score": 0.0,
+                        "_source": {
+                            "parent": "/1-default",
+                            "directory": "/1-default/sample1",
+                            "mtime": "2021-05-01T24:00:00",
+                            "mode": "0o755",
+                            "name": "sample1",
+                            "ancestor_path_elements": ["1-default"],
+                            "files": [
+                                {
+                                    "name": "result.txt",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 0,
+                                    "mode": "0o644",
+                                    "type": "reg",
+                                },
+                                {
+                                    "name": "user-benchmark.cmd",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 114,
+                                    "mode": "0o755",
+                                    "type": "reg",
+                                },
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                ],
+            },
+        }
+        index = self.build_index_from_metadata()
+
+        # get_expected_status() expects to read username and access from the
+        # JSON client payload, however this API acquires that information
+        # from the Dataset. Construct a fake payload corresponding to the
+        # attach_dataset fixture.
+        auth_json = {"user": "drb", "access": "private"}
+        expected_status = self.get_expected_status(
+            auth_json, build_auth_header["header_param"]
+        )
+
+        response = query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            self.payload,
+            index,
+            expected_status,
+            json=response_payload,
+            status=HTTPStatus.OK,
+            headers=build_auth_header["header"],
+        )
+        if expected_status == HTTPStatus.OK:
```

Comment:
The `if` isn't necessary: if the actual status hadn't matched `expected_status` then the `query_api` fixture would already have failed the test case with an assertion. (And despite having read that code many times, coming out of a busy weekend, my brain immediately jumped on this to say "ah ha; this would pass silently if the status wasn't OK.") Which is to say, doing the (redundant) check here could be misleading to an "unaware" reader. 😆 

You probably did this here because it was done elsewhere, but we'll worry about that some other time!

Reply:
And ... my mistake. I'm sorry. I'd completely forgotten that even though you're not using parameterized arguments here, the `query_api` fixture does that automatically. That means the "expected" status isn't always `OK`, and I got you to try to compare the output even when it's just an error message.

I think that's what I get for doing a review first thing in the morning. __Sorry__ about that!

Reply:
Sure. Thank you. yeah, that was related to Jenkins.. the environment was failing completely. but when I deleted some earlier tests and restarted the build it worked :)